examples/molecular_dynamics/example_molecular_dynamics.cpp

```diff
@@ -0,0 +1,178 @@
+      KOKKOS_LAMBDA(int i, int j, int k) {
+        int const id = i * ny * nz + j * nz + k;
+        particles[4 * id + 0] = {i * dx + .0f, j * dy + .0f, k * dz + .0f};
+        particles[4 * id + 1] = {i * dx + .0f, j * dy + .5f, k * dz + .5f};
+        particles[4 * id + 2] = {i * dx + .5f, j * dy + .0f, k * dz + .5f};
+        particles[4 * id + 3] = {i * dx + .5f, j * dy + .5f, k * dz + .0f};
+      });
+
+  Kokkos::View<float **, MemorySpace> velocities(
+      Kokkos::view_alloc("velocities", Kokkos::WithoutInitializing), n, 3);
+  { // scope so that random number generation resources are released
+    using RandomPool = Kokkos::Random_XorShift64_Pool<ExecutionSpace>;
+    RandomPool random_pool(5374857);
+
+    Kokkos::parallel_for(
+        "assign_velocities",
+        Kokkos::RangePolicy<ExecutionSpace>(execution_space, 0, n),
+        KOKKOS_LAMBDA(int i) {
+          RandomPool::generator_type generator = random_pool.get_state();
+          velocities(i, 0) = generator.frand(0.f, 1.f);
+          velocities(i, 1) = generator.frand(0.f, 1.f);
+          velocities(i, 2) = generator.frand(0.f, 1.f);
+          random_pool.free_state(generator);
+        });
+  }
+  // TODO scale velocities
+  Kokkos::Profiling::popRegion();
+
+  ArborX::BVH<MemorySpace> index(execution_space, particles);
+
+  Kokkos::View<int *, MemorySpace> indices("indices", 0);
+  Kokkos::View<int *, MemorySpace> offsets("offsets", 0);
+  index.query(execution_space, Neighbors<MemorySpace>{particles, r}, indices,
+              offsets);
+
+  Kokkos::View<float * [3], MemorySpace> forces("forces", n);
+  Kokkos::parallel_for(
+      "compute_forces",
+      Kokkos::RangePolicy<ExecutionSpace>(execution_space, 0, n),
+      KOKKOS_LAMBDA(int i) {
+        auto const x_i = particles(i)[0];
+        auto const y_i = particles(i)[1];
+        auto const z_i = particles(i)[2];
+        float fxi = 0.f;
+        float fyi = 0.f;
+        float fzi = 0.f;
+        for (int j = offsets(i); j < offsets(i + 1); ++j)
+        {
+          auto const dx = x_i - particles(indices(j))[0];
+          auto const dy = y_i - particles(indices(j))[1];
+          auto const dz = z_i - particles(indices(j))[2];
+          auto const rsq = dx * dx + dy * dy + dz * dz;
+          auto const cutoff_sq_ij = KokkosExt::ArithmeticTraits::infinity<
+              std::remove_cv_t<decltype(rsq)>>::value; // FIXME
+          if (rsq < cutoff_sq_ij)
+          {
+            auto const r2inv = static_cast<decltype(rsq)>(1) / rsq;
+            auto const r6inv = r2inv * r2inv * r2inv;
+            auto const lj1_ij = 1.f; // FIXME
+            auto const lj2_ij = 1.f; // FIXME
+            auto const fij = (r6inv * (lj1_ij * r6inv - lj2_ij)) * r2inv;
+            fxi += dx * fij;
+            fyi += dy * fij;
+            fzi += dz * fij;
+          }
+        }
+        forces(i, 0) += fxi;
+        forces(i, 1) += fyi;
+        forces(i, 2) += fzi;
+      });
+
+  float potential_energy;
```

Comment:
```suggestion
  float potential_energy = 0.f;
```

Reply:
Good point - energy and forces should be zeroed every step

Reply:
This is purely an output variable for the parallel reduce not an initial value.  This is why I did not initialize.